verbit/streaming_client.py

```diff
@@ -190,57 +203,148 @@ def _connect_websocket(self, media_config: MediaConfig, response_types: Response
         :param response_types: a bitmask Flag denoting which response type(s) should be returned by the server
         """
 
-        # build websocket url
+        # build WebSocket url
         ws_url = self.ws_url + self._get_ws_connect_query_string(media_config=media_config, response_types=response_types)
 
-        try:
+        # create WebSocket instance
+        self._ws_client = WebSocket(enable_multithread=True)
+
+        # set WebSocket client timeout
+        self._ws_client.timeout = 20.0
+
+        def _retry_http_error_predicate(retry_ex: WebSocketBadStatusException):
+
+            # specific 4xx Client-Errors trigger retry
+            if retry_ex.status_code in self.RETRY_HTTP_4XX_CODES:
+                return True
+
+            # don't retry all other 4xx client-errors
+            if 400 <= retry_ex.status_code < 500:
+                return False
```

Comment:
do we really know what hides here?
isn't it easier to list all retry codes and just assume everything else is no-retry?
why have 4 different cases like you did here?

Reply:
@roman-verbit-ai it's possible to list all the retry error codes, but I think it makes sense to have a separate default retry behavior for 4xx errors and for 5xx errors.

Reply:
Indeed let's comment on this, I took this 'common practice' from somewhere, I might add a ref,

The question of whether to retry - is whether the request is formulated correctly,
the rule of thumb: 
4xx: are 'client errors' - the except for '429' which is 'rate limiting', the servers opinion that the client's has issued too many requests, so 429 should not change the request
5xx: are 'server errors', basically we should differentiate  between permanent and temporary ones...